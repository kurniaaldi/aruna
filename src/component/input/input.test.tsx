import { render, screen } from "@testing-library/react";
import Input from "./index";
import renderer from "react-test-renderer";

test("input", () => {
  let values = {
    name: "input1",
    value: 10,
  };

  const handleChange = (event: any) => {};

  render(
    <Input value={values.value} name={values.name} onChange={handleChange} />
  );

  const inputElement = screen.getByTestId("input1");
  expect((inputElement as HTMLInputElement).value).toBe("10");
  expect((inputElement as HTMLInputElement).name).toBe("input1");
});

test("input-snapshoot", () => {
  let values = {
    name: "input1",
    value: 10,
  };

  const handleChange = (event: any) => {
    return event;
  };

  const tree = renderer
    .create(
      <Input value={values.value} name={values.name} onChange={handleChange} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
