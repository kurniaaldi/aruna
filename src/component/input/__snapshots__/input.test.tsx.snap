// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input-snapshoot 1`] = `
<input
  data-testid="input1"
  name="input1"
  onChange={[Function]}
  style={
    Object {
      "padding": 5,
      "width": "100%",
    }
  }
  type="number"
  value={10}
/>
`;
